### DISCUSSION TITLE
What actions in the lifecycle should be sent?

### DISCUSSION THREAD

**Discussion by dazy1030** [pointfreeco/swift-composable-architecture#2126] [Q&A] [answered]
Hi,  I have a rudimentary question.

I understand that the event that occurs is to update the state by notifying the Action in the send method of the ViewStore. So my question is, what kind of Action should I send when I want to do something in the lifecycle? For example, let's say I want to send a log and play a video by onAppear on a screen that plays a video. In this case, would the Action to be notified be "onAppear"? Or should it be "sendLog" and "play"?
I ask this question because there was a small sample using lifecycles.

Thank you.

**Comment by Alex293** [comment, marked as answer]
According to https://github.com/pointfreeco/swift-composable-architecture/discussions/1666#discussioncomment-4146758, onAppear or something similar should be preferred. Then in the reducer you can start both effects.

**Reply by dazy1030** [reply]
Oh, I had missed that... Thanks for letting me know! Now I will sleep well!
